app/controllers/spotlight/concerns/application_controller.rb

```diff
@@ -13,7 +13,11 @@ module ApplicationController
       helper Spotlight::ApplicationHelper
 
       rescue_from CanCan::AccessDenied do |exception|
-        redirect_to main_app.root_url, :alert => exception.message
+        if current_exhibit and !can? :read, current_exhibit
+          redirect_to main_app.new_user_session_url, :alert => exception.message
```

Comment:
We can't send them home here. If the exhibit is unpublished and they don't have access, they can't see the home page either.